src/userobjects/OpenMCTallyEditor.C

```diff
@@ -0,0 +1,144 @@
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2024 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#ifdef ENABLE_OPENMC_COUPLING
+
+#include "OpenMCTallyEditor.h"
+#include "OpenMCProblemBase.h"
+#include "openmc/tallies/tally.h"
+
+registerMooseObject("CardinalApp", OpenMCTallyEditor);
+
+InputParameters
+OpenMCTallyEditor::validParams()
+{
+  InputParameters params = GeneralUserObject::validParams();
+  params.addParam<bool>("create_tally", false, "Whether to create the tally if it doesn't exist");
+  params.addRequiredParam<int32_t>("tally_id", "The ID of the tally to modify");
+  params.addRequiredParam<std::vector<std::string>>("scores", "The scores to modify in the tally");
+  params.addRequiredParam<std::vector<std::string>>("nuclides",
+                                                    "The nuclides to modify in the tally");
+  params.addRequiredParam<std::vector<std::string>>("filter_ids",
+                                                    "The filter IDs to modify in the tally");
+  params.addParam<bool>(
+      "multiply_density", true, "Whether to multiply the tally by the atom density");
+  params.declareControllable("scores");
+  params.declareControllable("nuclides");
+  params.declareControllable("filter_ids");
+  params.declareControllable("multiply_density");
+  params.addClassDescription("A UserObject for creating and managing OpenMC tallies");
+  return params;
+}
+
+OpenMCTallyEditor::OpenMCTallyEditor(const InputParameters & parameters)
+  : OpenMCUserObject(parameters), _tally_id(getParam<int32_t>("tally_id"))
+{
+  this->initialize();
+}
+
+void
+OpenMCTallyEditor::initialize()
+{
+  // this is placed here, instead of the constructor, because Cardinal initializes
+  // some tallies. Depending on the order of initialization of UserObjects vs.
+  // other classes, those tallies might not exist yet in OpenMC's data space
+  // (but they will by the time this method is called).
```

Comment:
Does this comment refer to both of lines 61 and 62, or just line 62?

Reply:
It refers to the entirety of the block following. I'll remove the antecedent in the comment for clarity.